uikit/views/src/main/java/com/sygic/maps/uikit/views/searchtoolbar/SearchToolbar.kt

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (c) 2019 Sygic a.s. All rights reserved.
+ *
+ * This project is licensed under the MIT License.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.uikit.views.searchtoolbar
+
+import android.content.Context
+import android.text.TextWatcher
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.Button
+import android.widget.EditText
+import android.widget.TextView
+import androidx.appcompat.widget.Toolbar
+import com.sygic.maps.uikit.views.R
+import com.sygic.maps.uikit.views.databinding.LayoutSearchToolbarInternalBinding
+
+/**
+ * A [SearchToolbar] can be used as input component to the search screen. It contains [EditText] input field, state
+ * switcher (MAGNIFIER or PROGRESSBAR) and clear [Button].
+ *
+ * TODO MS-5681
+ *
+ */
+@Suppress("unused", "MemberVisibilityCanBePrivate")
+open class SearchToolbar @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyleAttr: Int = R.attr.searchToolbarStyle,
+    defStyleRes: Int = R.style.SygicSearchToolbarStyle // TODO: MS-5681
+) : Toolbar(context, attrs, defStyleAttr) {
+
+    private val binding: LayoutSearchToolbarInternalBinding =
+        LayoutSearchToolbarInternalBinding.inflate(LayoutInflater.from(context), this, true)
+
+    fun setText(text: String) {
+        binding.inputEditText.text?.let {
```

Comment:
tu ak by `getText` vratil `null`, nikdy by si nesetol novu hodnotu... neviem ci sa to realne moze stat, ale ak je to nullable 🤔

Reply:
pravdu máš, nemělo by to asi nastat, ale pro jistotu